src/blockdev.rs

```diff
@@ -195,6 +208,18 @@ pub fn reread_partition_table(file: &mut File) -> Result<()> {
     Ok(())
 }
 
+/// Get the logical sector size of a block device.
+pub fn get_sector_size(file: &mut File) -> Result<u32> {
+    let fd = file.as_raw_fd();
+    let mut size: c_int = 0;
+    match unsafe { blksszget(fd, &mut size) } {
+        Ok(_) => Ok(size
+            .try_into()
+            .chain_err(|| "sector size doesn't fit in u32")?),
```

Comment:
I don't expect this case to happen except for very weird cases, where I guess it could be helpful to display `size` somewhere in the error.